src/pages/Work/Layout.tsx

```diff
@@ -1,26 +1,33 @@
 import * as React from 'react';
 
-import { Overview } from './Overview';
-import { WorkTabs } from './WorkTabs';
+import { WorkById } from 'components/atoms/Work'
+import { Api } from 'helpers/PoetApi'
+import { Overview } from './Overview'
+import { WorkTabs } from './WorkTabs'
 
 import './Layout.scss';
 
-export interface WorkProps {
-  readonly id: string,
-  readonly historical: boolean
-}
-
-export class WorkLayout extends React.Component<WorkProps, undefined> {
+export class WorkLayout extends WorkById {
```

Comment:
It looks like the only reason this component is a class is to set `document.title` in `componentWillMount`.

1. Have you tried Next.js? It includes a head tag that lets you set things like the document title (and other header tags) inside JSX.

2. `componentDidMount()` is a more appropriate place to reference browser-specific APIs like `document`, because `componentWillMount()` runs on the server side when you set up universal rendering, which can cause it to throw errors in Node.

Reply:
Got it. Definitely want a functional approach to setting the document's header.

There's another reason for WorkLayout being a class: `extends WorkById`.  This is a hackish system Esteban and I introduced over a year ago and never got to replacing.

It has the nicety that you can associate React Components to URLs and forget about how the resource should be downloaded or whether it should be downloaded at all (it has a cache). But it forces you to use inheritance, which is against recommendations, and has the horrible side effect of having to use `renderElement` instead of `render`.

We'll need to do a deep refactor of our front ends when we can.

Reply:
Oh, I didn't notice that. Extending from your own components is a HUGE anti-pattern in React. That should be refactored to a higher-order component.

Class hierarchies are problematic, and create creeping, growing problems over time. Can you refactor it now?